pkg/pillar/types/downloadertypes.go

```diff
@@ -114,6 +114,19 @@ func (status DownloaderStatus) Pending() bool {
 	return status.PendingAdd || status.PendingModify || status.PendingDelete
 }
 
+// SetErrorInfo : Set Error Information for DownloaderStatus
+func (status *DownloaderStatus) SetErrorInfo(errStr string) {
+	status.RetryCount++
```

Comment:
Retrycount should be in the caller, since it is logically separate from setting errors. Even if all the callers increase it.

Reply:
Sure, have extracted the coumulative downloader fail hander functionality as HandleDownloadFail() method for DownloaderStatus.